### ISSUE TITLE
Program failed: Access violation in stack frame 5 at address 0x200005ff8 of size 8

### ISSUE THREAD

**Issue by retrofinger** [raydium-io/raydium-cp-swap#28]
After compiling and deploying on testnet, initializing amm_config, trying to create a pool using the included tests, i get the above error.

**Comment by retrofinger** [comment]
So i tried the same on devnet, just to be sure the problem is not with Testnet or the VM on it. But i get the same error.

Please note:

1.) I clone this repo
2.) I run anchor build in it
3.) I deploy it using anchor deploy
4.) I run: yarn run ts-mocha -p ./tsconfig.json -t 1000000 tests/initialize.test.ts
5.) The initialize test script calls the createAmmConfig function successfully
6.) The initialize test script fails to create the pool in the first test

Note: The amm_config account is being created correctly. I confirmed this.

Here are more details:

```
Message: Transaction simulation failed: Error processing Instruction 0: Program failed to complete. 
Logs: 
[
  "Program AywUGda6ZMcRHMt9XqEdvZ1L8eY3zFfsKGLv7xG57WSq invoke [1]",
  "Program log: Instruction: Initialize",
  "Program AywUGda6ZMcRHMt9XqEdvZ1L8eY3zFfsKGLv7xG57WSq consumed 6029 of 200000 compute units",
  "Program AywUGda6ZMcRHMt9XqEdvZ1L8eY3zFfsKGLv7xG57WSq failed: Access violation in stack frame 5 at address 0x200005ff8 of size 8"
]. 
Catch the `SendTransactionError` and call `getLogs()` on it for full details.
```

**Under this program id (AywUGda6ZMcRHMt9XqEdvZ1L8eY3zFfsKGLv7xG57WSq) you can find deployments of cp-swap on testnet and on devnet**. 

Interestingly, when compiling your program straight from the repo here, i get this error:

```
# anchor build
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /dev/raydium-cp-swap/programs/cp-swap/Cargo.toml
workspace: /dev/raydium-cp-swap/Cargo.toml
Error: Function _ZN112_$LT$solana_program..instruction..InstructionError$u20$as$u20$solana_frozen_abi..abi_example..AbiEnumVisitor$GT$13visit_for_abi17h43ecb57923db73a1E Stack offset of 4608 exceeded max offset of 4096 by 512 bytes, please minimize large stack variables
   Compiling raydium-cp-swap v0.1.0 (/dev/raydium-cp-swap/programs/cp-swap)
    Finished release [optimized] target(s) in 3.28s
```

Are the two errors related? If yes, how can the problem be fixed in the program? And if not, what am i doing wrong, what am i missing? Shouldn't the test scripts work with the repo as is? 

I'd very much like to deploy CP-SWAP on testnet so that i can run tests with jito (who doesn't support devnet, ironically). This is why i am trying to make it work.

**Comment by retrofinger** [comment]
So in case anyone else stumbles down this ugly rabbithole that wasted away multiple days of my time: DO NOT, i repeat **DO NOT** change the toolchain versions as they appear in Anchor.toml as it comes from the repo. I did that because **anchor build** refused to work with solana-program 1.17.0 as it was listed. I changed it to 1.18.25 and then things worked, but that lead to the stack error above. I downgraded to 1.17 and got my environment to work right and voila: it works.
